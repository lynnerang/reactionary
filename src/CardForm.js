import React, { Component } from 'react';
import './CardForm.scss';

const validType = file => (/\.(jpe?g|png|gif)$/i.test(file.name));

class CardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
    }
  }

  saveCard = (e) => {
    const term = document.querySelector('#my-term-input').value;
    const desc = document.querySelector('#my-desc-input').value;
    const file = document.querySelector('#my-example-input').value;

    this.props.saveNewCard(term, desc, file);
    document.querySelectorAll('.new-card-input').forEach(i => i.value = '');
  }

  getImage = (e) => {
    const reader = new FileReader();
    if (e.target.files[0] && validType(e.target.files[0])) {
      reader.readAsDataURL(e.target.files[0]); 
      // reader.onload = function() {document.querySelector('#my-example-input').value = e.target.result};
    } 
  }

  render() {
    return (
      <form className='new-card-form'>
        <label htmlFor='my-term-input'>Term<span>*</span>:</label>
        <input type='text' className='new-card-input' id='my-term-input' maxLength='30'></input>
        <label htmlFor='my-desc-input'>Description<span>*</span>:</label>
        <textarea className='new-card-input' id='my-desc-input' maxLength='80'></textarea>
        <label htmlFor='my-desc-input'>Example:</label>
        <input type='file' className='new-card-input' id='my-example-input' onChange={this.getImage}></input>
        <div className='new-card-form-btns'>
          <button type='button' className='cancel-card-btn' onClick={this.props.cancelForm}>Cancel</button>
          <button type='button' className='save-card-btn' onClick={this.saveCard}>Save</button>       
        </div>
      </form>
    )
  }
}

export default CardForm;