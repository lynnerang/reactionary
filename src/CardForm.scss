@import './variables.scss';

.new-card-form {
  display: flex;
  flex-direction: column;
  border: 1px solid $secondary-color;
  border-radius: 5px;
  padding: 20px;
  height: 380px;
  width: 335px;

  label {
    font-size: 18px;

    span {
      color: red;
    }
  }

  input,
  textarea {
    width: 290px;
    margin-bottom: 25px;
    margin-top: 5px;
    border-radius: 3px;
    outline-color: $secondary-color;
    min-height: 25px;
    padding: 10px;
    font-size: 18px;
  }

  #my-desc-input {
    height: 80px;
    border-color: rgb(196, 196, 196);
    border-top: 2px solid rgb(148, 148, 148);
    border-left: 2px solid rgb(148, 148, 148);
  }

  #my-example-input {
    margin-bottom: 0;
  }

  .new-card-form-btns {
    display: flex;
    align-self: flex-end;

    button {
      margin-top: 15px;
      margin-left: 15px;
      border-radius: 5px;
      padding: 5px 10px;
    }

    .save-card-btn {
      background: $secondary-color;
      border: none;
      color: white;
  
      &:hover {
        background: $blue;
      }
    }

    .cancel-card-btn:hover {
      color: darkred;
      border-color: darkred;
    }
  }
}