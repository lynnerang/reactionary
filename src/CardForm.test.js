import React from 'react';
import CardForm from './CardForm.js';
import { shallow } from 'enzyme';

const mockData = {
  term: 'oh',
  desc: 'hi',
  file: ''
}

const props = {
  saveNewCard: jest.fn(),
}


describe('CardForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow (
      <CardForm 
      />
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no default state properties', () => {
    expect(wrapper.state()).toEqual({});
  });

  it('can get input details and call saveNewCard with them.', () => {
    wrapper.instance().saveCard();

    expect(props.saveNewCard).toHaveBeenCalledTimes(1);

    expect(props.saveNewCard).toHaveBeenCalledWith(mockData.term, mockData.desc, mockData.file);
  });

});