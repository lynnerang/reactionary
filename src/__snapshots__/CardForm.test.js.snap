// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardForm Should match the snapshot 1`] = `
<form
  className="new-card-form"
>
  <label
    htmlFor="my-term-input"
  >
    Term
    <span>
      *
    </span>
    :
  </label>
  <input
    className="new-card-input"
    id="my-term-input"
    maxLength="30"
    type="text"
  />
  <label
    htmlFor="my-desc-input"
  >
    Description
    <span>
      *
    </span>
    :
  </label>
  <input
    className="new-card-input"
    id="my-desc-input"
    maxLength="80"
    type="textarea"
  />
  <label
    htmlFor="my-desc-input"
  >
    Example:
  </label>
  <input
    className="new-card-input"
    id="my-example-input"
    onChange={[Function]}
    type="file"
  />
  <div
    className="new-card-form-btns"
  >
    <button
      className="cancel-card-btn"
      type="button"
    >
      Cancel
    </button>
    <button
      className="save-card-btn"
      onClick={[Function]}
      type="button"
    >
      Save
    </button>
  </div>
</form>
`;
